include/convex_bodies/hpolytope.h

```diff
@@ -384,48 +386,49 @@ class HPolytope{
 
     // compute intersection point of a ray starting from r and pointing to v
     // with polytope discribed by A and b
-    std::pair<NT, int> line_positive_intersect(Point &r, Point &v, std::vector<NT> &Ar,
-            std::vector<NT> &Av) {
+    std::pair<NT, int> line_positive_intersect(Point &r, Point &v, VT& Ar,
+            VT& Av) {
         return line_intersect(r, v, Ar, Av, true);
     }
 
 
     // compute intersection point of a ray starting from r and pointing to v
     // with polytope discribed by A and b
-    std::pair<NT, int> line_positive_intersect(Point &r, Point &v, std::vector<NT> &Ar,
-            std::vector<NT> &Av, const NT &lambda_prev) {
+    std::pair<NT, int> line_positive_intersect(Point &r, Point &v, VT& Ar,
+            VT& Av, const NT &lambda_prev) {
         return line_intersect(r, v, Ar, Av, lambda_prev, true);
     }
 
 
     //First coordinate ray intersecting convex polytope
     std::pair<NT,NT> line_intersect_coord(Point &r, const unsigned int &rand_coord,
-                                          std::vector<NT> &lamdas) {
+                                          VT& lamdas) {
 
         NT lamda = 0, min_plus = NT(maxNT), max_minus = NT(minNT);
-        NT sum_nom, sum_denom;
+        VT sum_denom;
         unsigned int j;
         int m = num_of_hyperplanes();
         viterator rit;
```

Comment:
do we need rit?

Reply:
No, it is a leftover from the previous implementation. I will remove it.